### PR TITLE
Fix crash on detach, updated API, prepared version 2.0

### PR DESCRIPTION
**Pull request by pglebocki**
Make sure these boxes are checked before submitting your PR - thank you!

- [ ] Pull request is being merged to the `develop` branch.
- [ ] There's a reference to an *Issue* which this solves.
- [ ] For new features a sample has been added to the sample app
illustrating this feature's behaviour.
- [ ] Commits are squashed to a single commit.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Changelog
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
+and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
+
+## [2.0]
+### Changed
+- **Breaking change:** Replaced "positive/nagative button" clicked listener by global listener which need to be implemented by host activity
```
Comment (zawadz88) [inline CHANGELOG.md]: Maybe

> Replaced "positive/nagative button" clicked listener by global listener which need to be implemented by host activity

->

> Replaced "positive/negative button" clicked listener by a global listener which needs to be implemented by the host activity

?

File: app-rating/src/main/java/com/stepstone/apprating/AppRatingDialog.kt [hunk 2]
```diff
@@ -83,10 +81,6 @@ class AppRatingDialog protected constructor(private val activity: FragmentActivi
             var windowAnimationResId: Int = 0
 
             var noteDescriptions: List<String>? = null
```
Comment (zawadz88) [inline app-rating/src/main/java/com/stepstone/apprating/AppRatingDialog.kt]: This list must be also `Serializable` so maybe make this an `ArrayList`?

File: app/src/main/java/com/stepstone/apprating/sample/SampleActivity.kt [hunk 3]
```diff
@@ -38,7 +38,7 @@ class SampleActivity : FragmentActivity() {
                 .setDescriptionTextColor(R.color.descriptionTextColor)
                 .setCommentTextColor(R.color.commentTextColor)
                 .setCommentBackgroundColor(R.color.colorPrimaryDark)
-                .setPositiveButtonClickedListener(listener)
+                //.setPositiveButtonClickedListener(listener)
```
Comment (zawadz88) [inline app/src/main/java/com/stepstone/apprating/sample/SampleActivity.kt]: Shouldn't it be removed?

Comment (pglebocki) [reply app/src/main/java/com/stepstone/apprating/sample/SampleActivity.kt]: nice catch :)

File: CHANGELOG.md (hunk 1 above)
Comment (pglebocki) [reply CHANGELOG.md]: right, thank for correcting my English :)

**Pull request merged**